posthog/hogql_queries/query_runner.py

```diff
@@ -808,6 +814,29 @@ def run(
 
         return fresh_response
 
+    def get_api_queries_concurrency_limit(self):
+        """
+        :return: None - no feature, 0 - rate limited, 1,3,<other> for actual concurrency limit
+        """
+
+        if not settings.EE_AVAILABLE:
+            return None
+
+        from ee.billing.quota_limiting import list_limited_team_attributes, QuotaLimitingCaches, QuotaResource
+
+        if self.team.api_token in list_limited_team_attributes(
+            QuotaResource.API_QUERIES, QuotaLimitingCaches.QUOTA_LIMITER_CACHE_KEY
+        ):
+            return 0
```

Comment:
So this will only really happen if they have a billing limit set, they hit it and we quota limit them, correct?

Reply:
Yes, but also if they reach the *free limit quota*.